apps/builder/app/builder/features/seo/site-settings.tsx

```diff
@@ -0,0 +1,210 @@
+          width: theme.spacing[34],
+          mx: theme.spacing[5],
+          px: theme.spacing[5],
+        }}
+      >
+        <Label htmlFor={ids.siteName}>Site Name</Label>
+        <InputField
+          id={ids.siteName}
+          value={props.value.siteName ?? ""}
+          onChange={(event) => handleChange("siteName")(event.target.value)}
+          placeholder="Current Site Name"
+          name="Name"
+          autoFocus
+        />
+      </Grid>
+
+      <Separator />
+
+      <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+        <Label htmlFor={ids.favicon} sectionTitle>
+          Favicon
+        </Label>
+        <Grid flow="column" gap={3}>
+          <Image
+            width={72}
+            height={72}
+            className={imgStyle()}
+            src={favIconUrl}
+            loader={imageLoader}
+          />
+
+          <Grid gap={2}>
+            <Text color="subtle">
+              Upload a 32 x 32 px image to display in browser tabs.
+            </Text>
+            <ImageControl
+              assetId={props.value.faviconAssetId ?? ""}
+              onAssetIdChange={handleChange("faviconAssetId")}
+            >
+              <Button id={ids.favicon} css={{ justifySelf: "start" }}>
+                Upload
+              </Button>
+            </ImageControl>
+          </Grid>
+        </Grid>
+      </Grid>
+
+      <Separator />
+
+      <Grid gap={2} css={{ mx: theme.spacing[5], px: theme.spacing[5] }}>
+        <Label htmlFor={ids.code} sectionTitle>
+          Custom Code
+        </Label>
+        <Text color="subtle">
+          Custom code and scripts will be added at the end of the &lt;head&gt;
+          tag to every page across the published site.
+        </Text>
+        <TextArea
+          id={ids.code}
+          rows={5}
+          autoGrow
+          maxRows={10}
+          value={props.value.code ?? ""}
+          onChange={handleChange("code")}
+        />
+      </Grid>
+      <div />
+    </Grid>
+  );
+};
+
+const SiteSettingsView = () => {
+  const [value, setValue] = useState(
+    pagesStore.get()?.meta ?? {
+      siteName: "",
+      faviconAssetId: "",
+      code: "",
+    }
+  );
+
+  const open = useStore($isSiteSettigsOpen);
+
+  const handleSave = useEffectEvent(() => {
+    serverSyncStore.createTransaction([pagesStore], (pages) => {
+      if (pages === undefined) {
+        return;
+      }
+
+      pages.meta = value;
+    });
+  });
+
+  return (
+    <Dialog
+      open={open}
+      onOpenChange={(isOpen) => {
+        handleSave();
+        $isSiteSettigsOpen.set(isOpen);
+      }}
+    >
+      <DialogContent
+        onBlur={handleSave}
+        // Left Aside panels (e.g., Pages, Components) use zIndex: theme.zIndices[1].
+        // For a dialog to appear above these panels, both overlay and content should also have zIndex: theme.zIndices[1].
+        css={{
+          zIndex: theme.zIndices[1],
+        }}
+        overlayCss={{
```

Comment:
this is quite nasty, one idea to work this around is to check if css.zIndex specifically is defined and pass it to overlay inside DialogContent

Reply:
to avoid having to pass overlayCss at all

Reply:
Imo this should not exists at all and caused by poor aside panel implementation. Here I think explicit css is better vs implicit as most css systems do not allow to check existance of specific properties passed from top (as its usually function or classnames) I dont want to stick on stitches implementation

Reply:
I agree, I think the fundamental problem is the composition, the way radix does this - it allows to not have these problems, but we created a higher level component that has the composition built-in. In theory that higher level component should have z-index hardcoded.

I also agree z-index shouldn't be needed but if we take z-index abstract as a need to add any CSS override, still the problem is composition